### DISCUSSION TITLE
moving average filter

### DISCUSSION THREAD

**Discussion by PhD-GOAT** [neuropsychology/NeuroKit#872] [Q&A] [answered]
Hi, some of the API that extract hrv features apply a moving average filter to denoise. However, I wonder if the size of the window is not necessary because neurokit does not visualize the peak signal.

**Comment by DominiqueMakowski** [comment]
Not sure I understand, could you be more specific?

**Reply by PhD-GOAT** [reply]
I mean, I'm asking if there's a preprocessing for the outlier, the splashing value. I was wondering because I couldn't see the algorithm to filter when using the nk.hrv_time function.

**Reply by DominiqueMakowski** [reply, marked as answer]
Oh you mean if the original input is cleaned. The HRV functions don't do it by themselves; they assume that it you are inputting NK outputs from other functions (where cleaning is performed) or that you know what you are doing. 

Within NK this cleaning is achieved via for instance, `nk.ecg_clean()` and `nk.ecg_peaks()` (that runs `nk.signal_fixpeaks()`).

**Reply by PhD-GOAT** [reply]
Aha, then your words go from ecg_clean() to ecg_segment() when preprocessing from raw data for ecg. And in the case of ppg, do have to do ppg_clean(), ppg_findpeaks()?

**Reply by danibene** [reply]
yes, or you can use `ecg_process()` and `ppg_process()`, respectively
